cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tether/TetherStore.java

```diff
@@ -0,0 +1,215 @@
+      fields.add(Fields.longField(StoreDefinition.TetherStore.LAST_CONNECTION_TIME_FIELD, System.currentTimeMillis()));
+      StructuredTable tetherTable = context.getTable(StoreDefinition.TetherStore.TETHER);
+      tetherTable.update(fields);
+    });
+  }
+
+  /**
+   * Deletes a peer
+   *
+   * @param peerName name of the peer
+   * @throws IOException if deleting the table fails
+   */
+  public void deletePeer(String peerName) throws IOException {
+    TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable capabilityTable = context.getTable(StoreDefinition.TetherStore.TETHER);
+      capabilityTable
+        .delete(Collections.singleton(Fields.stringField(StoreDefinition.TetherStore.PEER_NAME_FIELD, peerName)));
+    });
+  }
+
+  /**
+   * Get information about all tethered peers
+   *
+   * @return information about status of tethered peer
+   * @throws IOException if reading from the database fails
+   */
+  public List<PeerInfo> getPeers() throws IOException {
+    List<PeerInfo> peers = new ArrayList<>();
+    return TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable tetherTable = context
+        .getTable(StoreDefinition.TetherStore.TETHER);
+      CloseableIterator<StructuredRow> iterator = tetherTable
+        .scan(Range.all(), Integer.MAX_VALUE);
+      iterator.forEachRemaining(row -> {
+        String peerName = row.getString(StoreDefinition.TetherStore.PEER_NAME_FIELD);
+        String endpoint = row.getString(StoreDefinition.TetherStore.PEER_URI_FIELD);
+        TetherStatus tetherStatus = TetherStatus.valueOf(row.getString(StoreDefinition.TetherStore.TETHER_STATE_FIELD));
+        PeerMetadata peerMetadata = GSON.fromJson(row.getString(StoreDefinition.TetherStore.PEER_METADATA_FIELD),
+                                                  PeerMetadata.class);
+        long lastConnectionTime = row.getLong(StoreDefinition.TetherStore.LAST_CONNECTION_TIME_FIELD);
+        peers.add(new PeerInfo(peerName, endpoint, tetherStatus, peerMetadata, lastConnectionTime));
+      });
+      return peers;
+    });
+  }
+
+  /**
+   * Get information about a peer
+   *
+   * @return information about status of a peer
+   * @throws IOException if reading from the database fails
+   * @throws PeerNotFoundException if the peer is not found
+   */
+  public PeerInfo getPeer(String peerName) throws IOException, PeerNotFoundException {
+    return TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable tetherTable = context
+        .getTable(StoreDefinition.TetherStore.TETHER);
+      Range range =  Range.singleton(
+        ImmutableList.of(Fields.stringField(StoreDefinition.TetherStore.PEER_NAME_FIELD, peerName)));
+      try (CloseableIterator<StructuredRow> iterator = tetherTable.scan(range, Integer.MAX_VALUE)) {
+        if (!iterator.hasNext()) {
+          throw new PeerNotFoundException(peerName);
+        }
+        StructuredRow row = iterator.next();
```

Comment:
This is repeated. Better to have a private method that takes a `StructuredRow` and return a `PeerInfo`

Reply:
Added private method getPeerInfo().